django/contrib/gis/db/models/fields.py

```diff
@@ -435,13 +434,11 @@ def contribute_to_class(self, cls, name, **kwargs):
         setattr(cls, self.attname, SpatialProxy(gdal.GDALRaster, self))
 
     def get_transform(self, name):
-        try:
+        with suppress(ValueError):
             band_index = int(name)
```

Comment:
Overall, the changes are mostly fine, I just wonder if we might use this as an opportunity to tighten up large try blocks and use try/except/else instead (such as here).